pyaedt/modules/PostProcessor.py

```diff
@@ -1122,6 +1122,52 @@ def available_report_quantities(
             )
         return []  # pragma: no cover
 
+    @pyaedt_function_handler()
+    def get_all_report_quantities(
+        self,
+        solution=None,
+        context=None,
+        is_siwave_dc=False,
+    ):
+        """Return all the possible report categories organized by report types, solution and categories.
+
+        Parameters
+        ----------
+        solution : str, optional
+            Report Setup.
+            The default is ``None``, in which case the all solutions are used.
+        context : str, dict, optional
+            Report Context.
+            The default is ``None``, in which case the default context is used.
+            For Maxwell 2D/3D Eddy Current solution types this can be provided as a dictionary
+            where the key is the matrix name and value the reduced matrix.
+        is_siwave_dc : bool, optional
+            Whether if the setup is Siwave DCIR or not. Default is ``False``.
+
+        Returns
+        -------
+        dict
+            A dictionary with primary key the report type, secondary key the solution type and
+            third key the report categories.
+        """
+        rep_quantities = {}
+        for rep in self.available_report_types:
+            rep_quantities[rep] = {}
+            if solution:
+                sols = [solution]
+            else:
+                sols = self.available_report_solutions(rep)
```

Comment:
Would it make sense to use an argument `solutions` instead of `solution` ? I mean, could a user want the report quantities for a set of solutions instead of one (or all if nothing is provided) ?

If yes, I would suggest to use `solutions` as an argument. That way, the behavior for one unique solution, multiples or none provided would be the same internally and we could get rid of `sols` and `sol` to use `solutions` and `solution`.

Reply:
Even if I cannot figure out why, it could be the case

Reply:
I've implemented it